pkg/controller/datavolume/import-controller.go

```diff
@@ -434,3 +487,72 @@ func (r *ImportReconciler) getNextCheckpoint(dataVolume *cdiv1.DataVolume, pvc *
 
 	return nil
 }
+
+func volumeImportSourceName(dv *cdiv1.DataVolume) string {
+	return fmt.Sprintf("%s-%s", volumeImportSourcePrefix, dv.UID)
+}
+
+func (r *ImportReconciler) createVolumeImportSourceCR(syncState *dvSyncState) error {
+	dv := syncState.dvMutated
+
+	source := &cdiv1.ImportSourceType{}
+	if http := dv.Spec.Source.HTTP; http != nil {
+		source.HTTP = http
+	} else if s3 := dv.Spec.Source.S3; s3 != nil {
+		source.S3 = s3
+	} else if gcs := dv.Spec.Source.GCS; gcs != nil {
+		source.GCS = gcs
+	} else if registry := dv.Spec.Source.Registry; registry != nil {
+		source.Registry = registry
+	} else if imageio := dv.Spec.Source.Imageio; imageio != nil {
+		source.Imageio = imageio
+	} else if vddk := dv.Spec.Source.VDDK; vddk != nil {
+		source.VDDK = vddk
+	} else {
+		// Our dv shouldn't be without source
+		// Defaulting to Blank source
+		source.Blank = &cdiv1.DataVolumeBlankImage{}
+	}
+
+	importSourceName := volumeImportSourceName(dv)
+	importSource := &cdiv1.VolumeImportSource{
```

Comment:
Totally forgot that the controller should be watching everything it creates, so should be watching `VolumeImportSources` owned by datavolumes.

Also, I think it would be better to see if the resource exists before creating, saves a trip to the apiserver, still have to handle already exists because cache is not perfect